eyelid-match-ops/src/primitives/yashe.rs

```diff
@@ -104,14 +118,71 @@ impl<C: PolyConf> Yashe<C> {
         (priv_key, pub_key)
     }
 
+    pub fn encrypt(&self, m: Message<C>, public_key: PublicKey<C>, rng: &mut ThreadRng) -> Ciphertext<C> {
+        let logt = self.params.t.ilog2();
+        let s = self.sample_err(rng);
+        let e = self.sample_err(rng);
+        let mut c = s * public_key.h + e;
+        // TODO: find a way to access q and divide by t
+        let mut qdt: BigInt<2> = BigInt::from(Coeff::MODULUS);
+        //let qbt = q / Coeff::from(self.params.t);
+        qdt.divn(logt);
+        let mqdt = m.m * Coeff::from(qdt);
```

Comment:
Normally you could multiply by `Coeff::inverse()` instead of dividing:
https://docs.rs/ark-ff/0.4.2/ark_ff/fields/trait.Field.html#tymethod.inverse

But that doesn't work for the modulus, because it reduces to zero. Instead, convert to `u128`, and do the arithmetic there.

I also added a `t_as_coeff()` method in PR #79, which could be useful here:
https://github.com/Inversed-Tech/eyelid/blob/3668934d68780513ea61ede8f4ccfb2d6a7eaedb/eyelid-match-ops/src/primitives/yashe/conf.rs#L36-L37

```suggestion
        // We can't divide the modulus using `Coeff` arithmetic, because it reduces to zero.
        let qdt = u128::from(q) / u128::from(t);
        let qdt = C::Coeff::from(qbt);
        let mqdt = m.m * qdt;
```

Reply:
It might be worth adding `inverse_t()` as a trait method after this method:
https://github.com/Inversed-Tech/eyelid/blob/3668934d68780513ea61ede8f4ccfb2d6a7eaedb/eyelid-match-ops/src/primitives/yashe/conf.rs#L36-L39

With this implementation:
```rust
        // We can't divide the modulus using `Coeff` arithmetic, because it reduces to zero.
        let qdt = u128::from(q) / u128::from(t);
        let qdt = C::Coeff::from(qbt);
```